File: backend/src/main/java/com/company/aiinterview/interview/entity/InterviewSession.java
package com.company.aiinterview.interview.entity;

import com.company.aiinterview.common.entity.BaseEntity;
import com.company.aiinterview.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "interview_sessions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InterviewSession extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "position_role")
    private String positionRole;

    @Column(name = "experience_level")
    private String experienceLevel;

    @Column(name = "interview_type")
    private String interviewType;

    private String language;

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "VARCHAR(50)")
    private InterviewStatus status;

    @Column(name = "current_turn")
    private Integer currentTurn;

    @Column(name = "target_turns")
    private Integer targetTurns;

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "ended_at")
    private LocalDateTime endedAt;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    @Column(name = "overall_ai_summary", columnDefinition = "TEXT")
    private String overallAiSummary;

    @Column(name = "overall_score")
    private Integer overallScore;

    @Column(name = "tech_stack", length = 500)
    private String techStack;

    @Column(name = "job_description", columnDefinition = "TEXT")
    private String jobDescription;

    @Column(name = "resume_text", columnDefinition = "LONGTEXT")
    private String resumeText;

    @Column(name = "is_hardcore", columnDefinition = "BOOLEAN DEFAULT FALSE")
    private Boolean isHardcore;
}

File: backend-node/src/models/InterviewSession.ts
import mongoose, { Document, Schema } from 'mongoose';

export enum InterviewStatus {
  PENDING = 'PENDING',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED'
}

export interface IInterviewSession extends Document {
  id: string;
  user: mongoose.Types.ObjectId;
  positionRole?: string;
  experienceLevel?: string;
  interviewType?: string;
  language?: string;
  status: InterviewStatus;
  currentTurn?: number;
  targetTurns?: number;
  startedAt?: Date;
  endedAt?: Date;
  durationSeconds?: number;
  overallAiSummary?: string;
  overallScore?: number;
  techStack?: string;
  jobDescription?: string;
  resumeText?: string;
  isHardcore: boolean;
  createdAt: Date;
  updatedAt: Date;
}

const interviewSessionSchema: Schema = new Schema(
  {
    user: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    positionRole: { type: String },
    experienceLevel: { type: String },
    interviewType: { type: String },
    language: { type: String },
    status: { type: String, enum: Object.values(InterviewStatus), default: InterviewStatus.PENDING },
    currentTurn: { type: Number, default: 0 },
    targetTurns: { type: Number, default: 5 },
    startedAt: { type: Date },
    endedAt: { type: Date },
    durationSeconds: { type: Number },
    overallAiSummary: { type: String },
    overallScore: { type: Number },
    techStack: { type: String },
    jobDescription: { type: String },
    resumeText: { type: String },
    isHardcore: { type: Boolean, default: false },
  },
  {
    timestamps: true,
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
);

export const InterviewSession = mongoose.model<IInterviewSession>('InterviewSession', interviewSessionSchema);
